QPrompt v2.0

Cuperino opened the pull request:
QPrompt v2.0.0 release.

Cuperino committed:
Bump to QPrompt 2.0

Cuperino committed:
Add network permissions

This allows using the "Open remote file" feature, introduced in the
unreleased version 1.2.

Cuperino committed:
Add .gitignore

ghost commented:
```
Fetching full git repo https://github.com/Cuperino/QPrompt.git
**fatal**: ambiguous argument '2.0.0-flatpak': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
Failed to download sources: module qprompt: Child process exited with code 128
error: Recipe `download-sources` failed with exit code 1
Error: Process completed with exit code 1.
```

Cuperino commented:
@yakushabb Sorry, it was supposed to be v2.0.0-flatpak. I just rebased with the fix a few moments ago.

ghost commented:
No problem. I'm also learning something new.

Cuperino commented:
Tested builds for both `x86_64` and `aarch64`. The aarch64 exhibits behaviors not present in any other build or package manager, suggesting problems with upstream Qt build. Both the x86_64 and aarch64 also exhibit a memory leak upon starting-up the app that is not present in any other build or packaging format.

I'll try reverting to using an earlier runtime-version in hopes to prevent these upstream bugs.

Cuperino committed:
Revert runtime-version bump to prevent memory leak and arm64 misbehavior

Cuperino committed:
Further lower runtime-version to prevent buttons misbehavior

Cuperino commented:
Lowering to 6.9 fixed the memory leak but the UI bugs continue.

Cuperino commented:
Neither the memory leak nor the UI bugs occur while using the 6.8 runtime.

Cuperino commented:
Files saved aren't saving to user-folders. A file saved to Documents instead goes to `/run/user/1000/doc/####/filename`...

ghost commented:
I think you need some permissions or use portals: https://docs.flatpak.org/en/latest/sandbox-permissions.html

Also you can your questions on Flathub discourse: https://discourse.flathub.org/

My knowledge is limited.

Cuperino committed:
Bump to v2.0.0 final

Cuperino committed:
Add missing filesystem permissions

Added permissions for Save As to work on the following folders:
- Documents
- Desktop
- Download
- Commonly used external storage mount-points

Cuperino commented:
@yakushabb Thanks, that helped. Apparently the write access is only needed for when saving files for the first time, such as when saving as.

As for the aforementioned UI bugs, I've fixed all the UI bugs that affected platforms 6.9 and 6.10 that could be fixed in QPrompt. With those fixes QPrompt is fully usable with up-to-date versions of Kirigami. However, some of the changes in Kirigami are also prevent Kirigami from working correctly with the Material theme that comes with Qt, which is used by QPrompt. These result in extra margins being added to the top and bottom of all buttons, which causes Kirigami's top menu to be vertically offset and a few padding issues which make the UI take-up more space than it should, impacting user productivity. For that reason, plus the memory leak in 6.10, I'll stick to the 6.8 platform until the top-menu displacement can be addressed upstream in Kirigami.

Cuperino commented:
With that I think QPrompt 2.0 is ready for reviewing. Feedback is welcome.

ghost commented:
I guess you have write permission to the repository? Otherwise, I will merge it when you are okay.

ghost commented:
There are warnings displayed during startup. 

> file:///usr/lib/qml/org/kde/kirigami/styles/Material/Theme.qml:85: TypeError: Cannot read property 'Kirigami' of null

Cuperino commented:
> file:///usr/lib/qml/org/kde/kirigami/styles/Material/Theme.qml:85: TypeError: Cannot read property 'Kirigami' of null

That warning in particular comes from Kirigami's code and occurs because we're using Kirigami with Qt's Material theme instead of KDE's Breeze or some other theme. Most of the warnings in QPrompt v2.0.0 are because of this. The app was in testing for over a year and received all the bug fixes I could address during that period. It works as intended despite the warnings. Every bug I could address in that time without breaking backwards compatibility got fixed.

> I guess you have write permission to the repository? Otherwise, I will merge it when you are okay.

It seems I'm able to merge, yes. In that case. I'll try to finish the Snap package and the new website first, then release everything at the same time. My plan is to release sometime tomorrow, on January 1st.
